internal/controller/api/job_receiver.go

```diff
@@ -1,23 +1,80 @@
 package api
 
 import (
+	"fmt"
+	"strings"
+
 	"github.com/gorilla/mux"
 
 	//	"context"
 	"encoding/json"
-	"io"
-	"io/ioutil"
 	"log"
 	"net/http"
 
 	"github.com/google/uuid"
 
 	"github.com/RedHatInsights/platform-receptor-controller/internal/controller"
+	"github.com/go-playground/validator/v10"
 	"github.com/redhatinsights/platform-go-middlewares/identity"
 
 	kafka "github.com/segmentio/kafka-go"
 )
 
+type badRequest struct {
+	status int
+	msg    string
+}
+
+func (br *badRequest) Error() string {
+	return fmt.Sprintf("%d: %s", br.status, br.msg)
+}
+
+func decodeJSON(w http.ResponseWriter, req *http.Request, job interface{}) error {
+	req.Body = http.MaxBytesReader(w, req.Body, 1048576)
+
+	dec := json.NewDecoder(req.Body)
+	dec.DisallowUnknownFields()
```

Comment:
I think it will be okay to just ignore unknown fields on the incoming json.  For now, I think we should be more forgiving and just ignore unknown fields.

Reply:
:+1: